wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Addresses.hs

```diff
@@ -51,10 +60,46 @@ newAddress NewAddress {..} = do
               >>= migrate
 
 -- | Verifies that an address is base58 decodable.
-verifyAddress :: Monad m => Text -> m (WalletResponse AddressValidity)
-verifyAddress address =
-    case decodeTextAddress address of
-        Right _ ->
-            return $ single $ AddressValidity True
-        Left _  ->
-            return $ single $ AddressValidity False
+getAddress
+    :: (MonadThrow m , V0.MonadWalletLogic ctx m)
+    => Text
+    -> m (WalletResponse WalletAddress)
+getAddress addrText = do
+    addr <- either
+        (throwM . InvalidAddressFormat)
+        pure
+        (decodeTextAddress addrText)
+
+    ss <- V0.askWalletSnapshot
+
+    let
+        addrId :: V0.CId V0.Addr
+        addrId =
+            V0.addressToCId addr
+
+        addrInfoMatchesAddr ::  V0.AddressInfo -> Bool
+        addrInfoMatchesAddr addrInfo =
+            addrId == V0.cwamId (V0.adiCWAddressMeta addrInfo)
+
+        getAddresses :: V0.CId V0.Wal -> [V0.AddressInfo]
+        getAddresses = V0State.getWAddresses ss V0.Ever
+
+        minfo :: Maybe (V0.CWalletMeta, V0.AddressInfo)
+        minfo =
+            asum
+            . map (\wid ->
+                (,) <$> V0State.getWalletMeta ss wid
+                    <*> List.find addrInfoMatchesAddr (getAddresses wid)
+                )
+            . V0.getWalletAddresses
+            $ ss
+
+    case minfo of
+        Nothing ->
+            throwM AddressNotFound
+        Just (_walletMeta, addressInfo) -> do
+            caddr <- V0.getWAddress
```

Comment:
Yeah, I'm not super pleased with it either. But since it is a legacy endpoint, we are going to throw it away soon anyway.

I figured out how to summon up that value, so we should be set there for change addresses (though we aren't currently returning that information, yet).